src/components/utilities/parseMclSettings.js

```diff
@@ -0,0 +1,6 @@
+import { defaultSettingsValues } from '../../constants';
+
+export default function parseMclSettings(settings, name, mcl) {
+  const parsedSettings = JSON.parse(settings?.records?.[0]?.value || '{}');
+  return parsedSettings?.[name]?.[mcl] ?? defaultSettingsValues?.[name]?.[mcl];
```

Comment:
If we are taking the above approach then the `name` would need to be replaced with `pageSize`